### PR TITLE
Improvement for MesosScheduler, cleanups and lint removal

### PR DESCRIPTION
**Pull request by briantopping**
In my deployment of 2.6.0-mr1-cdh5.4.0, I had a problem where "2015-05-08 03:33:24,421 INFO org.apache.hadoop.mapred.MesosScheduler: Unknown/exited TaskTracker: http://10.211.55.16:50060. " was being emitted 3-5 times a second. I don't have a JWDP connection figured out to see what the messages were, but I'm guessing there is an internal housekeeping job that is used internally. This implementation allows that to happen quietly.

* Restructure o.a.h.mapred.MesosScheduler#assignTasks to always pass through tasks, then decide whether they are a Mesos-managed TaskTracker
* ResourcePolicy is abstract for all intents, but it could be instantiated. Make it literally abstract
* A bunch of lint warnings removed
* Rearrange code to be easier to read -- interface implementations commented and methods in order, update some docs to JavaDoc format.
* Update POM version to 0.1.1-SNAPSHOT for additional work.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
         <!-- runtime deps versions -->
         <commons-logging.version>1.1.3</commons-logging.version>
         <commons-httpclient.version>3.1</commons-httpclient.version>
+        <!--<hadoop-client.version>2.6.0-mr1-cdh5.4.0</hadoop-client.version>-->
```
Comment (tarnfeld) [inline pom.xml]: Mind removing this from the PR?

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>org.apache.mesos</groupId>
     <artifactId>hadoop-mesos</artifactId>
-    <version>0.1.0</version>
+    <version>0.1.1-SNAPSHOT</version>
```
Comment (tarnfeld) [inline pom.xml]: Would also be good to remove the `SNAPSHOT` from the final release

File: src/main/java/org/apache/hadoop/mapred/MesosExecutor.java [hunk 3]
```diff
@@ -190,6 +189,7 @@ public void revokeSlots() {
     // Be sure there's nothing running and nothing in the launcher queue.
 
     // If we expect to have no slots, let's go ahead and terminate the task launchers
+    // CONDITION IS ALWAYS TRUE
```
Comment (tarnfeld) [inline src/main/java/org/apache/hadoop/mapred/MesosExecutor.java]: Feel free to remove both the `if()` statements wrapping this. Partly left around from a previous implementation that is unlikely to be resurrected.

**Comment by tarnfeld** [comment]
Thanks for the contributions! Nice to see the code being tidied up. I've made a couple of comments and once those are done i'll merge it and tag a release.

File: pom.xml (hunk 2 above)
Comment (tarnfeld) [inline pom.xml]: Actually, would you mind removing the version bump from this PR? We'll bump the version in master as there's another PR to be released, too.

Comment (briantopping) [inline pom.xml]: This is the first step to getting into a release process with Maven. It's a bit counterintuitive at first, ```SNAPSHOT``` means "the versions that lead up to the release without the 'SNAPSHOT' classifier". So versions leading up to 0.1.1 will be 0.1.1-SNAPSHOT, users should expect them to be evolving in nature. The release plugin will take care of removing ```SNAPSHOT``` when it's time to release. I'll set that up in a follow-on PR.

File: src/main/java/org/apache/hadoop/mapred/MesosExecutor.java (hunk 3 above)
Comment (briantopping) [inline src/main/java/org/apache/hadoop/mapred/MesosExecutor.java]: Great. After doing so, it was pretty obvious that the code could be simplified considerably :+1:

**Comment by briantopping** [comment]
Thanks for the time to review this so quickly! It's always a pleasure to contribute when PRs don't go stale. :)

Please give me a bit to get the release plugin set up. I'll document how to use it and we can try a release from there, sound good?

**Comment by tarnfeld** [comment]
Sounds great!

**Comment by briantopping** [comment]
Can I go ahead and set up artifact publishing to Sonatype Maven repository? This isn't an official Apache project, so we can't release it there.

**Comment by tarnfeld** [comment]
I wonder if the folk over at Mesosphere would be kind enough to host the project in their repos? They have a public maven repo over at http://downloads.mesosphere.io/maven. I'll ask around!

**Comment by briantopping** [comment]
Oh wait, my bad:

```
    <groupId>org.apache.mesos</groupId>
    <artifactId>hadoop-mesos</artifactId>
    <version>0.1.1-SNAPSHOT</version>
```

I can get this into the Apache repo if that's okay with the PMC.

**Comment by tarnfeld** [comment]
Maybe send an email to the dev mailing list, or ping the IRC? #mesos

**Comment by briantopping** [comment]
Pushing directly to Sonatype is the generally accepted clean path for small projects. If it's an Apache project, it should be pushed to the Apache repos. Both go directly to Maven Central. 

If this isn't an Apache project, we should use a different GAV (group/artifact/version) combo. 

WDYT?

**Comment by tarnfeld** [comment]
I'm not sure if this is _now_ technically an Apache project. It probably has that GAV because it was originally part of the Mesos source code and was brought out into it's own project by @brndnmtthws some time ago.

I think sending a message to the mailing list would be worth it, there's a bit of a wider discussion around this anyway related to hosting frameworks under `/mesos` on github (and if we're continuing to do that, my vote goes for hosting it in the Apache repos).

**Comment by briantopping** [comment]
Sent: http://www.mail-archive.com/dev@mesos.apache.org/msg32398.html

**Comment by tarnfeld** [comment]
Awesome. How would you feel about us taking this release configuration into a new PR and just keeping this one for the code cleanups/fixes?

**Comment by briantopping** [comment]
Sounds like an excellent idea. I've rolled back the version change per your request.

**Pull request merged**